test/misc.jl

```diff
@@ -572,6 +572,28 @@ let
     end
 end
 
+# Test that `print_with_color` accepts non-string values, just as `print` does
+let
+    old_have_color = Base.have_color
+    try
+        @eval Base have_color = true
+        buf_color = IOBuffer()
+        args = (3.2, "foo", :testsym)
+        print_with_color(:red, buf_color, args...)
+        buf_plain = IOBuffer()
+        print(buf_plain, args...)
+        string_color = String(take!(buf_color))
+        start_len = length(Base.text_colors[:red])
+        end_len = length(Base.text_colors[:default])
+        # Strip out the color control messages
```

Comment:
check that they match, not just that something of the same length happens to be there